tip.lic

```diff
@@ -0,0 +1,99 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#tip
+=end
+
+# Background: I've always been annoyed by the need to specify tips in coppers.
+# DR's required syntax leaves you counting zeroes if you want to tip someone.
+# So a tip of 10 platinum kronars requires you to tip 100000 kronars. How many
+# zeroes is that? I lost count already. This simple script allows you to tip 
+# someone, in a similar syntax, but use the denomination to do so (e.g. "platinum").
+# <;tip gildaren 10 plat kron> now works. If you don't specify a currency,
+# this script also chooses the currency of the greatest coin you have on hand.
+
+custom_require.call %w[common common-money]
+
+class Tip
+
+  # include DRC for the message function
+  # include DRCM for use of converting amounts to copper
+  
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'The recipent of your tip. Abbreviations ok.' },
+        { name: 'amount', regex: /^\d+$/i, variable: true, description: 'Numeric amount to tip, no commas (e.g. the number "100" in 100 kronars)' },
+        { name: 'denomination', regex: /(\bp(l|la|lat|lati|latin|latinu|latinum)?\b)|(\bg(o|ol|old)?\b)|(\bs(i|il|ilv|ilve|ilver)?\b)|(\bb(r|ro|ron|ronz|ronze)?\b)|(\bc(o|op|opp|oppe|opper)?\b)/i, variable: true, description: 'The denomination. Abbrevations ok. (e.g. plat, gold, silv)' },
+        { name: 'currency', regex: /(\bk(r|ro|ron|rona|ronar|ronars)?\b)|(l(i|ir|iru|irum|irums)?\b)|(\bd(o|ok|oko|okor|okora|okoras)?\b)/i, optional: true, variable: true, description: 'Optional currency. Abbreviations ok: kron, lirums, dokor' }         
+      ]
+    ]
+    args = parse_args(arg_definitions)  
+    gratuity(args.recipient, args.amount, args.denomination, args.currency)
+
+  end
+
+  def gratuity(recipient, amount, denomination, currency)
+    
+    total = convert_total(amount, denomination)
+    currency = handle_currency(total, currency)
+    
+    fput("tip #{recipient} #{total} #{currency}")
+    
```

Comment:
Tipping requires the other party to accept/reject/time out. Would it be desirable for this script to wait until the "tip process" has completed?

Reply:
At the moment I sort of have this designed to be an active script, meaning you're there using it live at the keyboard. Essentially just a quality of life script so we're not having to multiply by 1000 and carry the 2 just to figure out how much to tip.

BUT, I could flesh this out to be a more robust tipping script if ya'll thought it would be useful. Before writing this script I took a brief look at, I think it was bankbot, to see how they do it. In that script they do handle waiting for the tip process to complete, I believe by setting flags (which I haven't learned yet, but it would be a good chance to). But I believe that's designed to be fully automated.

Thoughts?

Reply:
I'd stick with your particular use case for now

Reply:
Maybe just note that it doesn't wait in the documentation or output a reminder to the user